src/odemis/acq/stream/_sync.py

```diff
@@ -1741,30 +1913,48 @@ def _runAcquisition(self, future):
             pos_flat = spot_pos.reshape((-1, 2))  # X/Y together (X iterates first)
             rep = self.repetition.value
             self._acq_data = [[] for _ in self._streams]  # just to be sure it's really empty
+            self._live_data = [[] for s in self._streams]
             self._raw = []
             self._anchor_raw = []
             logging.debug("Starting e-beam sync acquisition with components %s",
                           ", ".join(s._detector.name for s in self._streams))
-
             tot_num = numpy.prod(rep)
 
             # initialize leeches
             leech_np, leech_time_ppx = self._startLeeches(px_time, tot_num, (rep[1], rep[0]))
 
+            self._im_needs_recompute = threading.Event()
+            self._live_update_period = 0.33 # s
+            self._liveImageThread = threading.Thread(target = self._image_thread,
+                                                     args = (weakref.ref(self), self._live_update_period),
+                                                     name = "Live update of the acquisition")
+            self._liveImageThread.daemon = True
+            self._liveImageThread.start()
+
             # number of spots scanned so far
             spots_sum = 0
             while spots_sum < tot_num:
+                px_idx = (spots_sum//rep[0], spots_sum%rep[0]) #current pixel index
                 # Acquire the maximum amount of pixels until next leech
-                npixels = min([np for np in leech_np if np is not None] +
-                              [tot_num - spots_sum])  # max, in case of no leech
-                n_y, n_x = leech.get_next_rectangle((rep[1], rep[0]), spots_sum, npixels)
-                npixels = n_x * n_y
+                self._npixels2scan = min([np for np in leech_np if np is not None] +
+                                         [tot_num - spots_sum] + [int(self._live_update_period // px_time + 1)]) # max, in case of no leech
+                #TODO K.K. update unit tests to test for:
+                # cornercase where leech is set far after update_interval and therefore a non rectangular block may be scanned
+                # case where leech and update interval run semi in phase/out of phase
+                n_y, n_x = leech.get_next_rectangle((rep[1], rep[0]), spots_sum, self._npixels2scan)
+                self._npixels2scan = n_x * n_y
```

Comment:
Indicate *why* you recompute it afterwards (ie, it might be smaller, to stay a rectangle)